### PR TITLE
Require correct `expectedHeld`, Do not give refunds

### PR DESCRIPTION
**Pull request by NiloCK**
A stricter version of `deposit`. Requires that `expectedHeld` and `held` are equal.

PR is work toward #1410.

Previously: `deposit` had guardrails that prevented users from "overfunding" their commitments to channels. If a user deposited into a channel whose holdings exceeded their expectation, the adjudicator would make adjustments to the incoming deposit and refund the difference to the caller.

Now: `deposit` simply rejects any deposits where existing holdings do not match expectation. Users still cannot over-deposit.

### CODE REVIEW

**Comment by geoknee** [comment]
I'm on board with this. Refunds are an unnecessary feature which I don't think we envisaged anyone actually using.

This seems simpler, safer, and preserves the reorg-protection property.

**Comment by NiloCK** [comment]
> I'm on board with this. Refunds are an unnecessary feature which I don't think we envisaged anyone actually using.
> 
> This seems simpler, safer, and preserves the reorg-protection property.

Will continue this branch then and use it as a base for the batch-deposit (which play nicest with this version anyhow).

File: nitro-protocol/contracts/MultiAssetHolder.sol [hunk 1]
```diff
@@ -41,37 +41,23 @@ contract MultiAssetHolder is IMultiAssetHolder, StatusManager {
...
-        emit Deposited(channelId, asset, amountDeposited, nowHeld);
-
-        if (asset == address(0)) {
-            // refund whatever wasn't deposited.
-            uint256 refund = amount - amountDeposited;
-            (bool success, ) = msg.sender.call{value: refund}(''); //solhint-disable-line avoid-low-level-calls
-            require(success, 'Could not refund excess funds');
-        }
+        holdings[asset][channelId] += amount;
+        emit Deposited(channelId, asset, amount, holdings[asset][channelId]);
```
Comment (geoknee) [inline nitro-protocol/contracts/MultiAssetHolder.sol:60]: Nit: I think we may be able to save some gas by reducing the number of storage reads? Currently there are 3, and I think in principle it should be do-able in 1. Might be worth an investigation (no need to hold up this PR though).

**Review by geoknee — APPROVED** [review]

File: nitro-protocol/contracts/MultiAssetHolder.sol (hunk 1 above)
Comment (NiloCK) [reply nitro-protocol/contracts/MultiAssetHolder.sol:60]: :+1: Tracked in #1410

**Pull request merged**